src/components/transactions/GroupedTxListItems/WillReplaceProvider.tsx

```diff
@@ -0,0 +1,53 @@
+import { createContext, useMemo, useState, type Dispatch, type ReactElement, type SetStateAction } from 'react'
+import type { Transaction } from '@gnosis.pm/safe-react-gateway-sdk'
+
+import { useAppSelector } from '@/store'
+import { selectPendingTxs } from '@/store/pendingTxsSlice'
+
+export const WillReplaceContext = createContext<{
+  willReplace: string[]
+  setWillExecute: Dispatch<SetStateAction<string | undefined>>
+}>({
+  willReplace: [],
+  setWillExecute: () => {},
+})
+
+// Used for striking through transactions that will be replaced
+export const WillReplaceProvider = ({
+  groupedListItems,
+  children,
+}: {
+  groupedListItems: Transaction[]
+  children: ReactElement
+}): ReactElement => {
+  const [willExecute, setWillExecute] = useState<string>()
+  const pendingTxs = useAppSelector(selectPendingTxs)
+
+  const willReplace = useMemo(() => {
```

Comment:
`willExecute` and `willReplace` sound like they describe boolean variables. I would suggest something like `pendingItem` and `disabledItems`.